src/lines/ORMPLine.sol

```diff
@@ -49,7 +44,7 @@ contract ORMPLine is BaseMessageLine, Application, LineLookup, Ownable2Step {
         IEndpoint(TRUSTED_ORMP).send{value: msg.value}(toChainId, _toLine(toChainId), encoded, params);
     }
 
-    function recv(address fromDapp, address toDapp, bytes calldata message) external {
+    function recv(address fromDapp, address toDapp, bytes calldata message) external onlyORMP {
         uint256 fromChainId = _fromChainId();
         require(_xmsgSender() == _fromLine(fromChainId), "!auth");
```

Comment:
Need to discuss. onlyORMP   _xmsgSender() == _fromLine(fromChainId)

Reply:
Yes, onlyORMP check `msg.sender`, `_msgSender` check fromLine.